plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java

```diff
@@ -408,6 +412,30 @@ else if (identity.getId() == TRINO_MERGE_PARTITION_DATA) {
                 deletePredicate);
     }
 
+    private TupleDomain<IcebergColumnHandle> getEffectivePredicate(
+            Schema tableSchema,
+            Map<Integer, Optional<String>> partitionKeys,
+            TupleDomain<IcebergColumnHandle> dynamicFilterPredicate,
+            TupleDomain<IcebergColumnHandle> unenforcedPredicate)
+    {
+        if (!dynamicFilterPredicate.isAll()) {
+            Set<IcebergColumnHandle> partitionColumns = partitionKeys.keySet().stream()
+                    .map(fieldId -> getColumnHandle(tableSchema.findField(fieldId), typeManager))
+                    .collect(toImmutableSet());
+            Supplier<Map<ColumnHandle, NullableValue>> partitionValues = memoize(() -> getPartitionValues(partitionColumns, partitionKeys));
+            if (!partitionMatchesPredicate(partitionColumns, partitionValues, dynamicFilterPredicate)) {
```

Comment:
@raunaqmorarka before stripping the partition predicates from the dynamic filter we check now whether the dynamic filter predicate does not match the partition. Thank you for raising the point related to the DF being effective for split pruning at the page source provider level as well.